### PR TITLE
[RUM-14619] Add setViewLoadingTime() public API

### PR DESCRIPTION
**Pull request by MaelNamNam**
## Motivation

Allow developers to manually report view loading time for complex async patterns (SPAs with deferred fetching, skeleton screens) where auto-detection is inaccurate.

Brings Browser SDK to parity with iOS/Android SDKs.

## Changes

New experimental public API: `DD_RUM.setViewLoadingTime()`

- Marked as `[Experimental]` via JSDoc annotation (no runtime feature flag required)
- Loading time = elapsed since current view start
- First call suppresses auto-detected loading time
- Last call always wins
- Pre-start buffering supported (calls before `init()` are drained with preserved timestamps)
- Works with `startView()` / `manuallyTrackViews` — timer resets for each new view
- Telemetry sends `feature: 'addViewLoadingTime'` only (browser omits `overwritten`, `no_view`, and `no_active_view` — kept optional in schema for mobile compat)

**Note:** The `enableExperimentalFeatures: ['set_view_loading_time']` runtime gate has been removed. The API is available without any flag, consistent with other experimental APIs (`startAction`, `stopAction`, `startResource`, `stopResource`).

**Upstream dep:** [rum-events-format#352](https://github.com/DataDog/rum-events-format/pull/352) has been merged, submodule updated to `5ddfd3b`, types regenerated, and `[MANUAL]` annotation removed from this PR. `AddViewLoadingTime` is now auto-generated from the schema.

## Test instructions

```bash
yarn test:unit --spec packages/rum-core/src/domain/view/trackViews.spec.ts
yarn test:unit --spec packages/rum-core/src/boot/rumPublicApi.spec.ts
yarn test:unit --spec packages/rum-core/src/boot/preStartRum.spec.ts
yarn test:e2e:init && yarn test:e2e -g "manual loading time"
```

## Checklist

- [x] Tested locally
- [ ] Tested on staging
- [x] Added unit tests for this change.
- [x] Added e2e/integration tests for this change.
- [ ] Updated documentation and/or relevant AGENTS.md file

### CODE REVIEW

File: packages/rum-core/src/domain/view/trackViews.ts [hunk 1]
```diff
@@ -378,6 +383,19 @@ function newView(
       customTimings[sanitizeTiming(name)] = relativeTime
       scheduleViewUpdate()
     },
+    addLoadingTime(callTimestamp?: TimeStamp, overwrite = false) {
+      if (endClocks) {
+        return { no_active_view: true as const }
```
Comment (thomas-lebeau) [inline packages/rum-core/src/domain/view/trackViews.ts]: **🔨 warning:** ‏These are not following the standard naming convention. we use camelCase and convert what need to be to pascal_case just before sending the event.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts [hunk 2]
```diff
@@ -74,5 +77,26 @@ export function trackCommonViewMetrics(
       commonViewMetrics.interactionToNextPaint = getInteractionToNextPaint()
       return commonViewMetrics
     },
+    setManualLoadingTime: (loadingTime: Duration) => {
+      const hadPreviousManual = isLoadingTimeSetManually
+      const hadAutoValue = !isLoadingTimeSetManually && commonViewMetrics.loadingTime !== undefined
+
+      // Stop auto-detection entirely on first manual call (CONTEXT decision)
```
Comment (thomas-lebeau) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts]: **❓ question:** ‏what does `(CONTEXT decision)` means?

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts [hunk 3]
```diff
@@ -74,5 +77,26 @@ export function trackCommonViewMetrics(
...
+      if (!isLoadingTimeSetManually) {
+        stopLoadingTimeTracking()
+      }
+
+      isLoadingTimeSetManually = true
+      commonViewMetrics.loadingTime = loadingTime
+
+      // Return overwrite source for Phase 2 telemetry
```
Comment (thomas-lebeau) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts]: **❓ question:** ‏what's Phase 2 telemetry?

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 4]
```diff
@@ -727,6 +743,20 @@ export function makeRumPublicApi(
...
+      const result = strategy.addLoadingTime(callTimestamp, options?.overwrite ?? false)
+
+      // For pre-start buffered calls, result is undefined so we emit best-guess default values.
+      const isResult = result && typeof result === 'object'
+      addTelemetryUsage({
+        feature: 'addViewLoadingTime',
+        no_view: false,
+        no_active_view: isResult && 'no_active_view' in result,
+        overwritten: isResult && 'overwritten' in result && result.overwritten !== false,
+      })
```
Comment (thomas-lebeau) [inline packages/rum-core/src/boot/rumPublicApi.ts:793]: **❓ question:** TelemetryUsage is meant to understand which API the customer are using and with which option when available. I don't think we should return run-time analysis like this.


```suggestion
      addTelemetryUsage({
        feature: 'addViewLoadingTime',
        overwrite
      })
```

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 5]
```diff
@@ -215,6 +216,20 @@ export interface RumPublicApi extends PublicApi {
...
   addTiming: (name: string, time?: number) => void
 
+  /**
+   * Manually set the current view's loading time.
+   *
+   * Call this method when the view has finished loading. The loading time is computed as the
+   * elapsed time since the view started. By default, the first call sets the loading time and
+   * subsequent calls are no-ops. Use `{ overwrite: true }` to replace a previously set value.
+   *
+   * See [Override RUM View Loading Time](https://docs.datadoghq.com/real_user_monitoring/browser/advanced_configuration/) for further information.
```
Comment (thomas-lebeau) [inline packages/rum-core/src/boot/rumPublicApi.ts]: **🔨 warning:** "Override RUM View Loading Time" does not exist at that link ‏

File: packages/core/src/domain/telemetry/telemetryEvent.types.ts [hunk 6]
```diff
@@ -532,11 +532,13 @@ export type TelemetryCommonFeaturesUsage =
 /**
  * Schema of browser specific features usage
  */
+// [MANUAL] AddViewLoadingTime added pending upstream rum-events-format schema sync
```
Comment (thomas-lebeau) [inline packages/core/src/domain/telemetry/telemetryEvent.types.ts]: **❓ question:** ‏I did not find any PR related to that in rum-event-format

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 7]
```diff
@@ -727,6 +743,20 @@ export function makeRumPublicApi(
       strategy.addTiming(sanitize(name)!, time as RelativeTime | TimeStamp | undefined)
     }),
 
+    addViewLoadingTime: monitor((options?: { overwrite?: boolean }) => {
+      const callTimestamp = timeStampNow()
+      const result = strategy.addLoadingTime(callTimestamp, options?.overwrite ?? false)
+
+      // For pre-start buffered calls, result is undefined so we emit best-guess default values.
+      const isResult = result && typeof result === 'object'
```
Comment (thomas-lebeau) [inline packages/rum-core/src/boot/rumPublicApi.ts]: **💬 suggestion:** ‏use Types to prevent runtime assertion

File: packages/core/src/domain/telemetry/telemetryEvent.types.ts (hunk 6 above)
Comment (MaelNamNam) [reply packages/core/src/domain/telemetry/telemetryEvent.types.ts]: Opened: https://github.com/DataDog/rum-events-format/pull/352

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts (hunk 2 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts]: Woops, sorry, it was an internal note reminding why we keep auto-detection running instead of disabling it. I removed it in in 1bb569eff.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts (hunk 3 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts]: Sorry again, it was a reference to our internal development phasing (telemetry wiring came after the core logic). I removed in 1bb569eff.

File: packages/rum-core/src/boot/rumPublicApi.ts (hunk 5 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/rumPublicApi.ts]: Removed the broken link in 1bb569eff.

File: packages/rum-core/src/boot/rumPublicApi.ts (hunk 4 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/rumPublicApi.ts:793]: Good point. I went a different direction though — I aligned the telemetry with the mobile SDK schema instead, which tracks runtime state (`no_view`, `no_active_view`, `overwritten`). This matches what iOS/Android already report for `AddViewLoadingTime`. See c25c762fd.

File: packages/rum-core/src/boot/rumPublicApi.ts (hunk 7 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/rumPublicApi.ts]: Kept the runtime assertion — it feeds the `no_active_view` and `overwritten` telemetry fields that are now aligned with the mobile SDK schema. See c25c762fd.

File: packages/core/src/domain/telemetry/telemetryEvent.types.ts (hunk 6 above)
Comment (MaelNamNam) [reply packages/core/src/domain/telemetry/telemetryEvent.types.ts]: Opened as https://github.com/DataDog/rum-events-format/pull/352. Moved `AddViewLoadingTime` to the common schema (shared with mobile) with `no_view`, `no_active_view`, `overwritten` fields.

File: packages/rum-core/src/domain/view/trackViews.ts (hunk 1 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/trackViews.ts]: Fixed in 46d9511ab — renamed to camelCase internally (`noActiveView`, `overwritten`), conversion to snake_case now happens at the `addTelemetryUsage` boundary in rumPublicApi.ts.

File: .gitignore [hunk 8]
```diff
@@ -33,3 +33,4 @@ playwright-report/
 # Claude Code local files
 *.local.md
 .claude/settings.local.json
+.planning/
```
Comment (bcaudan) [inline .gitignore]: **💬 suggestion:** ‏as everyone could choose different locations for where they store plans, it could be preferable to configure a global gitignore file with your own preferences rather than adding it to all the repositories you contribute to.

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 9]
```diff
@@ -771,6 +792,15 @@ export function makeRumPublicApi(
       strategy.addTiming(sanitize(name)!, time as RelativeTime | TimeStamp | undefined)
     }),
 
+    setViewLoadingTime: monitor((options?: SetViewLoadingTimeOptions) => {
+      const callTimestamp = timeStampNow()
+      const result = strategy.setLoadingTime(callTimestamp, options?.overwrite ?? false)
+      addTelemetryUsage({
+        feature: 'addViewLoadingTime',
+        overwritten: result?.overwritten ?? false,
```
Comment (bcaudan) [inline packages/rum-core/src/boot/rumPublicApi.ts]: **❗ warning:** ‏‏preStartStrategy does not return anything so `result` is always undefined if the API is called before init.

cf https://github.com/DataDog/rum-events-format/pull/352#issuecomment-3998963616, unless we really want to monitor the actual behavior of this feature, I'd be in favor of: 
- only monitoring the API and API option usage
- drop the code computing and returning the overwritten result

File: sandbox/set-view-loading-time.html
Comment (bcaudan) [inline sandbox/set-view-loading-time.html:1]: **💬 suggestion:** ‏We usually don't commit specific feature sandbox files, it would be a burden to maintain.

File: packages/rum-core/src/domain/view/trackViews.ts [hunk 10]
```diff
@@ -376,6 +380,20 @@ function newView(
...
+      if (hasManualLoadingTime && !overwrite) {
+        return { overwritten: false }
+      }
+      const overwritten = hasManualLoadingTime
+      const loadingTime = elapsed(startClocks.timeStamp, callTimestamp ?? timeStampNow())
+      setManualLoadingTime(loadingTime)
+      hasManualLoadingTime = true
+      scheduleViewUpdate()
+      return { overwritten }
+    },
```
Comment (bcaudan) [inline packages/rum-core/src/domain/view/trackViews.ts]: **💬 suggestion:** scattering the logic between `trackViews` and `commonViewMetrics` make things harder to understand (in an already quite complex spot 😬)
We should probably:
- move all the needed logic to `commonViewMetrics`, as it is where loading time logic is already orchestrated
- enrich the exposed `setViewEnd` to notify the end of the view for both manual loading time and INP

File: .gitignore (hunk 8 above)
Comment (MaelNamNam) [reply .gitignore]: Good point. It's done

File: sandbox/set-view-loading-time.html
Comment (MaelNamNam) [reply sandbox/set-view-loading-time.html:1]: Noted, thanks. Removed in 712ed76e4.

File: packages/rum-core/src/domain/view/trackViews.ts (hunk 10 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/trackViews.ts]: Done. All manual loading time business logic (state, guards, elapsed calculation) moved into 
`trackCommonViewMetrics`. `trackViews.setLoadingTime` is now a 3-line forwarder, and `setViewEnd` is enriched to handle both INP and manual loading time end-of-view notification. See commits be0997573 and def5efcae.

File: packages/rum-core/src/boot/rumPublicApi.ts (hunk 9 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/rumPublicApi.ts]: Fixed!

File: packages/core/src/domain/telemetry/telemetryEvent.types.ts [hunk 11]
```diff
@@ -948,21 +950,15 @@ export interface StopResource {
 }
 export interface AddViewLoadingTime {
```
Comment (mormubis) [inline packages/core/src/domain/telemetry/telemetryEvent.types.ts]: Since you're already hand-editing this block under `[MANUAL]`, worth renaming the interface to `SetViewLoadingTime` to match the feature string (`'set-view-loading-time'`) and the public API (`setViewLoadingTime`). `AddViewLoadingTime` is a leftover from the old name.

File: packages/rum-core/src/boot/preStartRum.spec.ts [hunk 12]
```diff
@@ -608,6 +608,33 @@ describe('preStartRum', () => {
...
+      clock.tick(10)
+      strategy.setLoadingTime()
+
+      clock.tick(20)
+      strategy.init(DEFAULT_INIT_CONFIGURATION, PUBLIC_API)
+
+      expect(setLoadingTimeSpy).toHaveBeenCalledOnceWith(jasmine.any(Number), false)
+      // Verify the timestamp was captured at call time (tick 10), not at drain time (tick 30)
+      const capturedTimestamp = setLoadingTimeSpy.calls.argsFor(0)[0] as number
+      expect(capturedTimestamp).toBeLessThan(clock.timeStamp(30))
```
Comment (mormubis) [inline packages/rum-core/src/boot/preStartRum.spec.ts]: This assertion is too loose — it would pass even if the timestamp was captured at t=29. A regression that deferred capture to just before drain time wouldn't be caught. `toBe(clock.timeStamp(10))` would actually pin down what the comment claims to verify.

File: packages/rum-core/src/boot/preStartRum.ts [hunk 13]
```diff
@@ -241,6 +241,10 @@ export function createPreStartStrategy(
       bufferApiCalls.add((startRumResult) => startRumResult.addTiming(name, time))
     },
 
+    setLoadingTime: ((callTimestamp = timeStampNow(), overwrite = false) => {
```
Comment (mormubis) [inline packages/rum-core/src/boot/preStartRum.ts]: This default never fires — `rumPublicApi` always passes an explicit timestamp. Misleads readers into thinking pre-start has its own independent capture. Either drop it or add a comment explaining it's just a defensive fallback.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts [hunk 14]
```diff
@@ -69,10 +74,28 @@ export function trackCommonViewMetrics(
     },
     stopINPTracking,
     setLoadEvent,
-    setViewEnd,
+    setViewEnd: (viewEndTime: RelativeTime) => {
+      viewEnded = true
+      inpSetViewEnd(viewEndTime)
```
Comment (mormubis) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts]: Nit: `setINPViewEnd` reads more naturally than `inpSetViewEnd`.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+import { createMutationRecord, mockGlobalPerformanceBuffer, mockRumConfiguration } from '../../../../test'
+import { PAGE_ACTIVITY_END_DELAY, PAGE_ACTIVITY_VALIDATION_DELAY } from '../../waitPageActivityEnd'
+import { LifeCycle } from '../../lifeCycle'
+import type { RumMutationRecord } from '../../../browser/domMutationObservable'
+import { trackCommonViewMetrics } from './trackCommonViewMetrics'
+
+const BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY = (PAGE_ACTIVITY_VALIDATION_DELAY * 0.8) as Duration
+const AFTER_PAGE_ACTIVITY_END_DELAY = PAGE_ACTIVITY_END_DELAY * 1.1
+
+describe('trackCommonViewMetrics', () => {
```
Comment (mormubis) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts:16]: Nit: a one-liner explaining why this spec exists separately from `trackViews.spec.ts` (e.g. "tests the internal module in isolation to avoid setting up the full view stack for edge cases") would help future readers land faster.

File: test/e2e/scenario/rum/views.scenario.ts [hunk 16]
```diff
@@ -66,6 +66,51 @@ test.describe('rum views', () => {
...
+              }, 500)
+            })
+        )
+
+        await flushEvents()
+        const viewEvent = intakeRegistry.rumViewEvents.at(-1)
+        expect(viewEvent).toBeDefined()
+        // Should reflect the second (overwritten) value (~500ms), not the first (~200ms)
+        expect(viewEvent!.view.loading_time).toBeGreaterThanOrEqual(500 * 1e6)
+      })
```
Comment (mormubis) [inline test/e2e/scenario/rum/views.scenario.ts:113]: Both new tests call `setViewLoadingTime()` after init. The buffered pre-start path (`setViewLoadingTime()` → `init()`) has unique timing behavior and is only covered at unit level — worth adding an E2E sanity check for it too.

File: packages/core/src/domain/telemetry/telemetryEvent.types.ts (hunk 11 above)
Comment (MaelNamNam) [reply packages/core/src/domain/telemetry/telemetryEvent.types.ts]: Woooooops, sorry, I missed this one. Thanks & renamed in 4bb65b3.

File: packages/rum-core/src/boot/preStartRum.spec.ts (hunk 12 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/preStartRum.spec.ts]: Thanks. I pinned it to `toBe(clock.timeStamp(10))` as you suggested in 4a6331e.

File: packages/rum-core/src/boot/preStartRum.ts (hunk 13 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/preStartRum.ts]: Got it, I removed the default so `callTimestamp` is now a required `TimeStamp` param. See bc37a25.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts (hunk 15 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts:16]: Done. I Added a one-liner in eba54d6.

File: test/e2e/scenario/rum/views.scenario.ts (hunk 16 above)
Comment (MaelNamNam) [reply test/e2e/scenario/rum/views.scenario.ts:113]: Thanks! It's done in 72c0bf3.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts (hunk 14 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.ts]: Much better, thanks, I renamed it to `setINPViewEnd` in a177d40

File: packages/rum-core/src/boot/preStartRum.ts (hunk 13 above)
Comment (MaelNamNam) [reply packages/rum-core/src/boot/preStartRum.ts]: And fixed the test in 75431bb 😅

File: packages/core/src/domain/telemetry/telemetryEvent.types.ts (hunk 11 above)
Comment (MaelNamNam) [reply packages/core/src/domain/telemetry/telemetryEvent.types.ts]: Actually, I went too fast on this and I'm reverting it (cf 00a28dc): the telemetry event is called `add-view-loading-time` because it was define as such in the Mobile SDKs. As this `setViewLoadingTime` is an experimental API, I feel it's best to re-use already existing event names rather than add a new one just for Browser that might eventually get removed.

File: packages/rum-core/src/domain/view/trackViews.ts [hunk 17]
```diff
@@ -376,6 +379,11 @@ function newView(
       customTimings[sanitizeTiming(name)] = relativeTime
       scheduleViewUpdate()
     },
+    setLoadingTime(callTimestamp?: TimeStamp, overwrite = false) {
+      if (setLoadingTimeOnMetrics(callTimestamp, overwrite)) {
+        scheduleViewUpdate()
+      }
+    },
```
Comment (bcaudan) [inline packages/rum-core/src/domain/view/trackViews.ts]: **💬 suggestion:** ‏what about scheduling view update from inside common view metrics?
it would:
- avoid this extra wrapper, `currentview.setLoadingTime` could directly be `commonViewMetrics.setLoadingTime`
- avoid to return a boolean from `commonViewMetrics.setLoadingTime`

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+  beforeEach(() => {
+    mockGlobalPerformanceBuffer()
+    clock = mockClock()
+    domMutationObservable = new Observable()
+    windowOpenObservable = new Observable()
+    scheduleViewUpdateSpy = jasmine.createSpy('scheduleViewUpdate')
+  })
+
+  describe('manual loading time suppresses auto-detected loading time callback', () => {
+    it('should ignore auto-detected loading time when manual loading time was already set', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts:32]: **❓ question:** ‏is it not already covered by `should suppress auto-detected loading time after manual call`?

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+      clock.tick(100)
+      const secondResult = setLoadingTime(undefined, true)
+      expect(secondResult).toBe(false)
+
+      expect(getCommonViewMetrics().loadingTime).toBe(firstValue)
+    })
+  })
+
+  describe('setLoadingTime overwrite logic', () => {
+    it('should reject setLoadingTime without overwrite after first call', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts]: **❓ question:** ‏is it not already covered by `should not overwrite loading time by default on subsequent calls`?

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+      const firstValue = getCommonViewMetrics().loadingTime
+
+      clock.tick(200)
+      const result = setLoadingTime()
+      expect(result).toBe(false)
+
+      expect(getCommonViewMetrics().loadingTime).toBe(firstValue)
+    })
+
+    it('should allow overwrite before view end', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts]: **❓ question:** ‏is it not already covered by `should overwrite loading time when overwrite is true`?

File: packages/rum-core/src/domain/view/trackViews.ts (hunk 17 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/trackViews.ts]: Updated in 85fa995. `setLoadingTime` now calls `scheduleViewUpdate` internally and no longer returns a boolean. The wrapper in `trackViews` is replaced by a direct alias.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts (hunk 19 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts]: Sorry, I missed this one. Removed in 85fa995.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts (hunk 20 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts]: Sorry and same, removed in 85fa995.

File: packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts (hunk 18 above)
Comment (MaelNamNam) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts:32]: Not entirely, this test covers a specific race condition where `setLoadEvent` fires after a manual loading time was already set, which triggers the auto-detection callback. The `trackViews` test doesn't exercise that path. Do you feel I should delete this test too?

Comment (bcaudan) [reply packages/rum-core/src/domain/view/viewMetrics/trackCommonViewMetrics.spec.ts:32]: No, let's keep it then

**Review by mormubis — APPROVED** [review]
LGTM, thanks for the changes!
